### PR TITLE
Feature #664 3.0 lib refactor

### LINKED ISSUE #664: 3.0 lib refactor
To move with pype 3.0, we need to refactor `lib.py` which is a mishmash of functions into a clean testable code inside a new lib python module.

The agreed plan is to split it into individual tasks. Everyone takes functions they wrote originally, judges where it needs to go, refactors, writes tests, if time allows. PRs with individual functions of groups of functions after refactor, should be made into `feature/664-3-0-lib-refactor ` branch. Once everything is pulling into it, we'll merge into develop. 

@kalisp will be reviewing smaller PRs to ensure consistency of code style, docstrings and general approach.

### PR DESCRIPTION
**Pull request by kalisp**
Contains multiple merged PR restructuring monolith of lib.py into separate files in `lib` folder.

Small grooming, no logical changes done.

Closes #664

### CODE REVIEW

File: pype/tests/test_lib_restructuralization.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        from pype.lib import get_linked_assets
+        from pype.lib import get_latest_version
+        from pype.lib import ffprobe_streams
+
+        from pype.hosts.fusion.lib import switch_item
+
+        from pype.lib import source_hash
+        from pype.lib import _subprocess
+
+    except ImportError as e:
```
[the remark this answers is not in the document]
Comment (kalisp) [reply pype/tests/test_lib_restructuralization.py:38]: Just simple import tests, imports are not supposed to be used later.

**Comment by kalisp** [comment]
Conflict is that 2.x/develop still contain lib.py that should be removed and then merged.

**Comment by mkolar** [comment]
Looking good, but the conflict needs to be resolved

**Comment by mkolar** [comment]
> Conflict is that 2.x/develop still contain lib.py that should be removed and then merged.

Better way is to merge 2.x/develop to this branch, Delete the lib.py and then this will be mergeable

**Comment by kalisp** [comment]
Conflict resolved

**Review by mkolar — APPROVED** [review]

**Pull request merged**